python/plugins/processing/modeler/ModelerAlgorithm.py

```diff
@@ -208,15 +208,9 @@ class ModelerAlgorithm(GeoAlgorithm):
     CANVAS_SIZE = 4000
 
     def getCopy(self):
-        newone = ModelerAlgorithm()
-        newone.provider = self.provider
-        newone.algs = copy.deepcopy(self.algs)
-        newone.inputs = copy.deepcopy(self.inputs)
+        newone = ModelerAlgorithm.fromFile(self.descriptionFile)
```

Comment:
This is not always possible. The algorithm might not be saved yet if it's run from the modeler instead of the toolbox